foreman/data_refinery_foreman/foreman/main.py

```diff
@@ -1334,46 +1335,51 @@ def cleanup_the_queue():
-            if index not in active_volumes:
-                # The index for this job isn't currently mounted, kill
-                # the job and decrement the retry counter (since it
-                # will be incremented when it is requeued).
-                try:
-                    nomad_client.job.deregister_job(job["ID"], purge=True)
-                    logger.debug('Foreman Killed nomad job because it had a volume that was not active',
-                                  nomad_job_id=job['ID'], job_type=job_type)
-                    processor_job = ProcessorJob.objects.filter(nomad_job_id=job["ID"]).first()
-
-                    if not processor_job:
-                        # If it's not a processor job, it's probably a downloader job.
-                        job_record = DownloaderJob.objects.filter(nomad_job_id=job["ID"])[0]
-                        # If it's a downloader job, then it doesn't
-                        # have to run on the volume it was assigned
-                        # to. We can let the foreman reassign it.
-                        job_record.volume_index = None
-                        job_record.save()
-
-                    num_jobs_killed += 1
-                except:
-                    logger.exception("Could not remove Nomad job from the Nomad queue.",
-                                     nomad_job_id=job["ID"],
-                                     job_type=job_type)
-                    # If we can't do this for some reason, we'll get it next loop.
-                    pass
+        # ensure the job is one of the indexed_job_types
+        job_type = None
+        for pipeline in indexed_job_types:
+            if job["ParentID"].startswith(pipeline):
+                job_type = pipeline
+                break
+        if not job_type:
+            continue
+
+        # If this job has an index, then its ParentID will
+        # have the pattern of <job-type>_<index>_<RAM-amount>
+        # and we want to check the value of <index>:
+        split_parent_id = job["ParentID"].split("_")
+        if len(split_parent_id) < 2:
+            continue
+        else:
+            index = split_parent_id[-2]
+
+        if index not in active_volumes:
+            # The index for this job isn't currently mounted, kill the job
+            # `num_retries` will be decremented when the job receives the SIGKILL
+            try:
+                nomad_client.job.deregister_job(job["ID"], purge=True)
+
+                if job_type == DOWNLOADER:
+                    job_record = DownloaderJob.objects.filter(nomad_job_id=job["ID"]).first()
+                    # If it's a downloader job, then it doesn't
+                    # have to run on the volume it was assigned
+                    # to. We can let the foreman reassign it.
+                    job_record.volume_index = None
+                    job_record.save()
+
+                logger.info('Foreman Killed nomad job because it did not have a volume assigned',
```

Comment:
Why does this say "did not have a volume assigned" instead of "volume that was not active" like it used to? Seems like it was more accurate before.

Reply:
I think I misunderstood what was happening, let me change it.

```suggestion
                logger.info('Foreman Killed nomad job because it had a volume that was not active',
```